src/node/node.c

```diff
@@ -356,16 +365,27 @@ static void set_value(struct s_plugin_conf *conf, const char *key,
 
 	if (dst_env == NULL) {
 		/* Allocate one */
-		if (conf->size == MAX_ENV_NB) {
+		if (conf->used == MAX_ENV_NB) {
 			fprintf(stderr, "ran out of internal env space\n");
 			abort();
 		}
+		if (conf->used == conf->size) {
+			conf->size = conf->size * 3 / 2;
+			if (conf->size == 0)
+				conf->size = 4;
+			if (conf->size > MAX_ENV_NB)
+				conf->size = MAX_ENV_NB;
+
+			conf->env =
+			    realloc(conf->env,
+				    sizeof(struct s_env) * conf->size);
+			if (conf->env == NULL)
+				oom_handler();
+		}
 
 		/* ptr arithmetic is done with int, not with size_t */
-		dst_env = conf->env;
-		dst_env += (int) conf->size;
-
-		conf->size++;
+		dst_env = conf->env + (int) conf->used;
+		conf->used++;
```

Comment:
I'm also fond of my [1SLOC paradigm](https://blog.pwkf.org/2022/09/18/always-optimize-for-dummies.html#1sloc---one-statement-per-line-of-code), as written in https://blog.pwkf.org/2022/09/18/always-optimize-for-dummies.html